jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/CatalogResource.java

```diff
@@ -118,9 +118,28 @@ public CatalogResource(final JaxrsUriBuilder uriBuilder,
     @Produces(APPLICATION_XML)
     @ApiOperation(value = "Retrieve the full catalog as XML", response = String.class, hidden = true)
     @ApiResponses(value = {})
-    public Response getCatalogXml(@javax.ws.rs.core.Context final HttpServletRequest request) throws Exception {
+    public Response getCatalogXml(@QueryParam(QUERY_REQUESTED_DT) final String requestedDate,
+                                  @javax.ws.rs.core.Context final HttpServletRequest request) throws Exception {
         final TenantContext tenantContext = context.createTenantContextNoAccountId(request);
-        return Response.status(Status.OK).entity(XMLWriter.writeXML((VersionedCatalog) catalogUserApi.getCatalog(catalogName, tenantContext), VersionedCatalog.class)).build();
+        final DateTime catalogDateVersion = requestedDate != null ?
+                                            DATE_TIME_FORMATTER.parseDateTime(requestedDate).toDateTime(DateTimeZone.UTC) :
+                                            null;
+
+        final VersionedCatalog catalog = (VersionedCatalog) catalogUserApi.getCatalog(catalogName, tenantContext);
+
+        String result = null;
+        if (catalogDateVersion != null) {
+            for (final StandaloneCatalog v : catalog.getVersions()) {
+                if (v.getEffectiveDate().compareTo(catalogDateVersion.toDate()) == 0) {
+                    result = XMLWriter.writeXML(v, StandaloneCatalog.class);
+                    break;
```

Comment:
The other issue is that this end point now returns different type of response based on the query param. I am wondering if it would not be cleaner to either have 2 separate endpoints or at least return the same type of response. A `VersionedCatalog` is pretty much a list of `StandaloneCatalog` so we could return one one element when query param is provided. 

@pierre Any thoughts?

Reply:
👍 for returning a `VersionedCatalog` with a single catalog in that case.